lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp

```diff
@@ -85,14 +85,9 @@ static bool IsSwiftAsyncFunctionSymbol(swift::Demangle::NodePointer node) {
     return false;
 
   // Peel off layers over top of Function nodes.
-  switch (node->getFirstChild()->getKind()) {
-  case Node::Kind::Static:
-  case Node::Kind::ExplicitClosure:
+  while (llvm::is_contained({Node::Kind::Static, Node::Kind::ExplicitClosure},
```

Comment:
Is this a quadratic algorithm?

Reply:
suggestion: only support nested `ExplicitClosure`.

as far as I know, there's no case where `Static` nodes are nested. While there's no harm in supporting nested `Static` nodes, I'd prefer to see the code match the reality. Purely for expressiveness. I'd rather not have a reader wonder "why does this need to handle nested `Static` nodes"?

Reply:
> Is this a quadratic algorithm?

It's linear in the depth of the tree: one node in each level of the tree is compared against two possibilities, i.e. it perform constant operations on each level of the tree.

> as far as I know, there's no case where Static nodes are nested.

So is it safe to say that if there is a Static node, it must be at the root of the tree?